pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/util/KitUtil.java

```diff
@@ -170,97 +170,110 @@ public class KitUtil {
-                            ElasticSearchUtil.BY_GUID + kitLabelTriggered.getDdpParticipantId());
-                    if (participantESData == null || participantESData.isEmpty()) {
-                        participantESData = ElasticSearchUtil.getFilteredDDPParticipantsFromES(ddpInstance,
-                                ElasticSearchUtil.BY_LEGACY_ALTPID + kitLabelTriggered.getDdpParticipantId());
-                    }
-                    DDPParticipant ddpParticipant =
-                            ElasticSearchUtil.getParticipantAsDDPParticipant(participantESData, kitLabelTriggered.getDdpParticipantId());
-                    if (ddpParticipant != null) {
-                        toAddress = KitRequestShipping.getToAddressId(easyPostUtil, kitLabelTriggered.getKitRequestSettings(), null,
-                                ddpParticipant, ddpInstanceDto);
-                        KitRequestShipping.updateRequest(kitLabelTriggered, ddpParticipant, kitLabelTriggered.getKitTyp(),
-                                kitLabelTriggered.getKitRequestSettings());
-                    } else {
-                        KitRequestShipping.deactivateKitRequest(Long.parseLong(kitLabelTriggered.getDsmKitRequestId()),
+                if (StringUtils.isBlank(kitLabel.getAddressIdTo())) {
+                    DDPParticipant ddpParticipant = getDDPParticipant(kitLabel.getDdpParticipantId(), ddpInstance);
+                    if (ddpParticipant == null) {
+                        KitRequestShipping.deactivateKitRequest(Long.parseLong(kitLabel.getDsmKitRequestId()),
                                 "Participant not found", null, SystemUtil.SYSTEM, ddpInstanceDto);
-                        logger.error("Didn't find participant " + kitLabelTriggered.getDdpParticipantId());
+                        logger.error("Did not find participant {} for kit {}",
+                                kitLabel.getDdpParticipantId(), kitLabel.getDsmKitId());
+                    } else {
+                        toAddress = KitRequestShipping.getToAddressId(easyPostUtil, kitLabel.getKitRequestSettings(),
+                                null, ddpParticipant, ddpInstanceDto);
+                        KitRequestShipping.updateRequest(kitLabel, ddpParticipant, kitLabel.getKitTyp(),
+                                kitLabel.getKitRequestSettings());
                     }
                 } else {
+                    // TODO: I believe this case does not exist, so setting this to trip alert if it ever does.
+                    // ideally we would remove this code or fix it - DC
+                    logger.error("Handling case where kit has an address id: assumed case would not occur: "
+                                   + "KitCreateLabel: {}", kitLabel);
                     //uploaded pt
-                    toAddress = KitRequestShipping.getToAddressId(easyPostUtil, kitLabelTriggered.getKitRequestSettings(),
```

Comment:
The changes to the next 40 lines are due to renaming a variable and fixing formatting.
As mentioned in the summary, my plan was to do a bigger refactor but that is now on hold.